### ISSUE TITLE
[Bug][test-string]: search and query use prefix expressions  report an  error"fail to search on all shard leaders, err=fail to Search, QueryNode ID=4, reason=Search 4 failed"

### ISSUE THREAD

**Issue by jingkl** [milvus-io/milvus#17046]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20220516-59bc0a70
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):2.1.0.dev57
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

query testcase
 ```
   @pytest.mark.tags(CaseLabel.L1)
    def test_query_string_expr_with_prefixes(self):
        """
        target: test query with 
        method: specify string primary field as output field
        expected: return string primary field
        """
        collection_w, vectors = self.init_collection_general(prefix, insert_data=True, primary_field=ct.default_string_field_name)[0:2]
        expression = "varchar startsWith \"0\""
        res, _ = collection_w.query(expression)
```

search testcase
```
     perfix_expr = "varchar startsWith \"0\""
     @pytest.mark.tags(CaseLabel.L2)
    def test_search_string_field_not_primary_perfix(self, auto_id, _async):
        """
        target: test search with string expr and string field is not primary
        method: create collection and insert data 
                create index and collection load
                collection search uses string expr in string field, string field is not primary
        expected: Search successfully
        """
        # 1. initialize with data
        collection_w, _, _, insert_ids = \
            self.init_collection_general(prefix, True, auto_id=auto_id, dim=default_dim)[0:4]
        # data = cf.gen_default_list_data()
        # # data =cf.gen_default_string_data(ct.default_nb)
        # collection_w.insert(data)
        # collection_w.load()
        # 2. search
        log.info("test_search_string_field_not_primary: searching collection %s" % collection_w.name)
        vectors = [[random.random() for _ in range(default_dim)] for _ in range(default_nq)]
        output_fields = [default_string_field_name, default_float_field_name]
        collection_w.search(vectors[:default_nq], default_search_field,
                            default_search_params, default_limit, 
                            perfix_expr,
                            output_fields=output_fields,
                            _async=_async,
                            travel_timestamp=0
                            )

```
log:
```
[2022-05-17 09:58:27 - ERROR - pymilvus.decorators]: RPC error: [_execute_search_requests], <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=4, reason=Search 4 failed, reason [UnexpectedError] unsupported operator type: 8 err %!w(<nil>))>, <Time:{'RPC start': '2022-05-17 09:58:26.975043', 'RPC error': '2022-05-17 09:58:27.319131'}> (decorators.py:73)
[2022-05-17 09:58:27 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=4, reason=Search 4 failed, reason [UnexpectedError] unsupported operator type: 8 err %!w(<nil>))>, <Time:{'RPC start': '2022-05-17 09:58:26.940436', 'RPC error': '2022-05-17 09:58:27.319302'}> (decorators.py:73)
[2022-05-17 09:58:27 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/jingjing/Workspace/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/Users/jingjing/Workspace/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 690, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/decorators.py", line 56, in handler
    raise e
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/decorators.py", line 41, in handler
    return func(self, *args, **kwargs)
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/decorators.py", line 21, in handler
    return func(self, *args, **kwargs)
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 499, in search
    return self._execute_search_requests(requests, timeout, **_kwargs)
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/decorators.py", line 74, in handler
    raise e
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/decorators.py", line 70, in handler
    return func(*args, **kwargs)
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 446, in _execute_search_requests
    raise pre_err
  File "/Users/jingjing/miniconda3/envs/bbb/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 437, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=4, reason=Search 4 failed, reason [UnexpectedError] unsupported operator type: 8 err %!w(<nil>))>
 (api_request.py:35)
[2022-05-17 09:58:27 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=4, reason=Search 4 failed, reason [UnexpectedError] unsupported operator type: 8 err %!w(<nil>))> (api_request.py:36)
```

**Comment by longjiquan** [comment]
Already fixed by #17050 . Please retry this case. thx.
/assign @jingkl

**Comment by jingkl** [comment]
pymilvus :2.1.0.dev57
milvus tag: master-20220517-b37b87eb
Already fixed
